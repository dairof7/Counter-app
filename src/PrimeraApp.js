//import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types'


//primera letra en mayuscula indica que es un componente
const PrimeraApp = ( {saludo='noname', subtitulo='Soy un subtitulo'} ) => {

    //const saludo = 'Hola ke ase';
    

    return (
    //aqui se puede usar un div, pero asi se evita llenar de divs innecesarios 
    
    <>
        {/* una forma de imprimir objetos en html
        <pre>{ JSON.stringify( saludo, null, 3 ) }</pre> */}
        <h1>{ saludo }</h1>
        <h2>{ subtitulo }</h2>
    </>
    );
} 
//de esta manera me aseguro que el tipo de entrada es del tipo deseado
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    //edad: PropTypes.number.isRequired

}

//otra forma de poner valores por default
//PrimeraApp.defaultProps = {
//    edad: 21
//}

/* //lo anterior tambien se puede hacer con un fragment
const PrimeraApp = () => {
    return (
    <Fragment>
        <h1>hola</h1>
        <h1>como estas</h1>
    </Fragment>
    );
} 
 */


export default PrimeraApp;