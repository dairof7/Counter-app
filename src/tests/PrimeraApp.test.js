//import { render } from '@testing-library/react'
import PrimeraApp from '../PrimeraApp'
import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'


describe('Prueba primeraApp componente', () => {

//esta prueba depende de extend-expect que esta comentado en setupTest.js
/* 
    test('debe mostrar el mensaje Hola, soy dairo', () => {
        const saludo = 'Hola, soy dairo';
        const { getByText } = render( <PrimeraApp saludo={ saludo } /> );
        expect( getByText(saludo) ).toBeInTheDocument();
    }) */
    

    test('debe mostrar <PrimeraApp /> correctamente ', () => {
        const saludo = 'Hola, soy dairo'
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );
        expect ( wrapper ).toMatchSnapshot()
    });

    test('debe mostrar el subtitulo enviado ', () => {
        const saludo = 'Hola, soy dairo'
        const sub = 'Soy un subtitulo'

        const wrapper = shallow( 
            <PrimeraApp
                saludo={ saludo }
                subtitulo={ sub }
            />
        );
        const textoSubtitulo = wrapper.find('h2').text();
        expect (sub).toBe(textoSubtitulo)
    })
})
