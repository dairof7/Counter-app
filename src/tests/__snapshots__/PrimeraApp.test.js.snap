// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba primeraApp componente debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy dairo
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
</Fragment>
`;
